docs/docs/dev_docs/contracts/syntax/storage.md

```diff
@@ -1,40 +1,533 @@
+
+#include_code remove /yarn-project/aztec-nr/aztec/src/state_vars/set.nr rust
+
+An example of how to use this operation is visible in the `easy_private_state`:
+
+#include_code remove /yarn-project/aztec-nr/easy-private-state/src/easy_private_state.nr rust
+
+### `get_notes`
+
+This function returns the notes the account has access to:
+
+#include_code get_notes /yarn-project/aztec-nr/aztec/src/state_vars/set.nr rust
+
+There's a limit on the maximum number of notes this function can return at a time. Check [here](https://github.com/AztecProtocol/aztec-packages/blob/master/yarn-project/aztec-nr/aztec/src/constants_gen.nr) and look for `MAX_READ_REQUESTS_PER_CALL` for the up-to-date number.
+
+Because of this limit, we should always consider using the second argument `NoteGetterOptions` to target the notes we need, and to reduce the time required to recursively call this function.
+
+An example of this is is using the [filter_notes_min_sum](https://github.com/AztecProtocol/aztec-packages/blob/master/yarn-project/aztec-nr/value-note/src/filter.nr) to get "enough" notes to cover a given value. 
+
+#include_code get_notes /yarn-project/aztec-nr/easy-private-state/src/easy_private_state.nr rust
+
+### `view_notes`
+Functionally similar to [`get_notes`](#get_notes), but executed unconstrained and can be used by the wallet to fetch notes for use by front-ends etc.
+
+#include_code view_notes /yarn-project/aztec-nr/aztec/src/state_vars/set.nr rust
+
+There's also a limit on the maximum number of notes that can be returned in one go. To find the current limit, refer to [this file](https://github.com/AztecProtocol/aztec-packages/blob/master/yarn-project/aztec-nr/aztec/src/constants_gen.nr) and look for `MAX_NOTES_PER_PAGE`.
+
+The key distinction is that this method is unconstrained. It does not perform a check to verify if the notes actually exist, which is something the [`get_notes`](#get_notes) method does under the hood. Therefore, it should only be used in an unconstrained contract function.
+
+This function requires a `NoteViewerOptions`:
+
+#include_code NoteViewerOptions /yarn-project/aztec-nr/aztec/src/note/note_viewer_options.nr rust
+
+The `NoteViewerOptions` is essentially similar to the [`NoteGetterOptions`](#notegetteroptions), except that it doesn't take a custom filter.
+
+### NoteGetterOptions
+
+`NoteGetterOptions` encapsulates a set of configurable options for filtering and retrieving a selection of notes from a data oracle. Developers can design instances of `NoteGetterOptions`, to determine how notes should be filtered and returned to the functions of their smart contracts.
```

Comment:
Can you say a bit more or link to more info about data oracles?

Reply:
Data oracles is essentially one of the oracles from the functions section, but can add a link in here